tiled/server/dependencies.py

```diff
@@ -160,19 +163,16 @@ def expected_shape(
 
 
 def slice_(
-    slice: str = Query(None, pattern="^[-0-9,:]*$"),
+    slice: Optional[str] = None,
```

Comment:
I would like to retain a regex pattern. It no longer has a security function, with `eval` removed, but it does have a role in both documenting the acceptable inputs in the OpenAPI spec and providing a detailed error message. (See my example in PR Conversation tab.)

Additionally, it would be useful to test that all the failing examples (typo and malicious) will not match the regex, and therefore should get bounced by FastAPI before they parsed by our code. That is, the `ValueError` or `TypeError` should never actually get raised when used in integration with FastAPI. 

Finally, the pattern I've used here is a little under-specified. It is sufficient to block malicious examples but not sufficient to block malformed (e.g. typo-ed) examples. A more precise regex was drafted [in these lines of a dormant PR](https://github.com/bluesky/tiled/pull/391/files#diff-d41273077415e8ba3ad05d118aaa7167ef40343ad385a84666ea1cdfda8454e7R122-R127). Perhaps that would be reused here (with the `mean` aspect removed for now).

Reply:
Thanks for the feedback! I'll write a commit with updates.

Reply:
I agree.  The rest looks good to me.